src/bsoncxx/CMakeLists.txt

```diff
@@ -46,6 +47,10 @@ elseif(BSONCXX_POLY_OPTIONS_SET EQUAL 0)
         # If we are in C++17 mode, use the C++17 versions
```

Comment:
Update the `foreach(BSONCXX_POLY_OPTION` above to include `${BSONCXX_POLY_USE_IMPLS}`.